### ISSUE TITLE
Cursor appears at unexpected location when at end of multiline command line who's length is window width -1

### ISSUE THREAD

**Issue by msftrncs** [PowerShell/PSReadLine#1216]
Environment data
----------------
Experienced on:
PowerShell 7.0.0-preview6, PSReadLine 2.0.0-beta6 + debug in ConsoleHost on Windows 10.

```
PS version: 7.0.0-preview.6
PSReadline version: 2.0.0-beta6
os: 10.0.19033.1 (WinBuild.160101.0800)
PS file version: 7.0.0.0
HostName: ConsoleHost
BufferWidth: 120
BufferHeight: 9001
```

Steps to reproduce
--------------------------------------
1. At a prompt, enter a command that is 1 character short of the full window width.  Note that the command prompt takes up some of the space.  The cursor should be in the last window column at this time.
2. Activate the command for InsertLineBelow, usually <kbd>SHIFT</kbd>+<kbd>ENTER</kbd>.
3. <kbd>UpArrow</kbd>

At this point the cursor will be in the first window column of the next line (in the continuation prompt!), instead of the last column of the first line.

![image](https://user-images.githubusercontent.com/26179051/69910458-762d6300-13d1-11ea-9ba8-eecf75502ba4.png)

**Comment by msftrncs** [comment]
I have a simple fix for this and #1217, but then I hit #1218, and I am trying to make sure its unrelated.

**Comment by msftrncs** [comment]
Furthermore, combine this with #1216, with the CJK character at the end scrolled to the next line, and now the cursor appears on the wrong line.

My cursor should be behind `hello`, not above it.
![image](https://user-images.githubusercontent.com/26179051/70022844-38f4dc80-155b-11ea-8d8c-b3459dee6d4f.png)
